File: result/translate/starchat/java2python/python/N_TH_NUMBER_WHOSE_SUM_OF_DIGITS_IS_TEN.py
def f_gold(n):
        count = 0
        for curr in range(1, 1000000):
            sum = 0
            while curr > 0:
                digit = curr % 10
                sum += digit
                curr //= 10
            if sum == 10:
                count += 1
            if count == n:
                return curr

File: result/translate/chatgpt/python2java/java/N_TH_NUMBER_WHOSE_SUM_OF_DIGITS_IS_TEN.java
public int f_gold(int n) {
    int count = 0;
    Iterator<Integer> iterator = new Iterator<Integer>() {
        int curr = 0;
        
        @Override
        public boolean hasNext() {
            return true;
        }

        @Override
        public Integer next() {
            int x = curr;
            int sum = 0;
            while (x > 0) {
                sum += x % 10;
                x /= 10;
            }
            if (sum == 10) {
                count++;
            }
            if (count == n) {
                return curr++;
            }
            return curr++;
        }
    };
  
    return -1;
}